6212 Иванов Максим Алексеевич Лаб. 1 Вар. 26

Gwyn1300 opened the pull request:
**ФИО:** Иванов Максим 
**Номер группы:** 6212  
**Номер лабораторной:** 1
**Номер варианта:** 26 
**Краткое описание решаемой задачи:** Согласно варианту 26 лабораторной 1 мне надо было привести все данные к единому шаблону (любому из допустимых). Сохранить результат в новый файл. Если данные указаны некорректно - поставить прочерк. 
**Краткое описание работы кода:** Модуль `file_operation.py` работает с файлом (считывает и создает новый). Модуль `main.py` является точкой входа в приложение. Модуль `validators.py` проверяет на корректность данных файла. Модуль `processors.py` обрабатывает данные. Чтобы его запустить, требуется указать аргументы командной строки: `python main.py -<путь к файлу с данными>` (Или указать в файле `settings.json` следующие параметры ...).

Gwyn1300 committed:
Выполнена 1 лабораторная работа

IlyaOv commented on lab1/file_operations.py:
```diff
@@ -0,0 +1,32 @@
+    ''' Записывает обработанные данные в новый файл.''' 
+    try:
+        directory = os.path.dirname(original_path)
+        new_file_path = os.path.join(directory, 'data_processed.txt')
```
Выходной файл тоже лучше передавать аргументом командной строки

IlyaOv commented on lab1/processors.py:
```diff
@@ -0,0 +1,107 @@
+    return "Город: -"
+
+
+def process_lines(lines: str) -> str:
```
В функции многократное повторение if i < len(lines):
Можно упростить.

IlyaOv reviewed (changes requested):
Код нужно привести к [PEP8](https://peps.python.org/pep-0008/)

Gwyn1300 committed:
Добавлен аргумент для выходного файла

Gwyn1300 committed:
Все файлы:
отформатирован по PEP8
file_operation.py:
write_processed_file: добавлено сохранение по заданному пути
processors.py:
process_lines:
упрощена

Gwyn1300 committed:
Исправлено форматирование

IlyaOv reviewed (approved):
Посмотрите как оформлять группы импортов. В остальном зачтено.
